lua/telescope/make_entry.lua

```diff
@@ -290,30 +306,45 @@ function make_entry.gen_from_quickfix(opts)
   end
 end
 
-function make_entry.gen_from_symbols(opts)
+function make_entry.gen_from_lsp_symbols(opts)
   opts = opts or {}
-  opts.tail_path = get_default(opts.tail_path, true)
+  local bufnr = opts.bufnr or vim.api.nvim_get_current_buf()
 
-    local displayer = entry_display.create {
-    separator = "",
+  local displayer = entry_display.create {
+    separator = " ",
     items = {
-      { width = 6 },
-      { width = 40 },
-      { width = 1 },
-      { remaining = true },
-      { width = 1 },
-    },
-  }
+      { width = 8 },
+      { width = 25 },
+      { width = 8 },
+      { remaining = true }
+      }
+    }
 
   local make_display = function(entry)
-    local filename
-    if not opts.hide_filename then
-      filename = entry.filename
-      if opts.tail_path then
-        filename = utils.path_tail(filename)
-      elseif opts.shorten_path then
-        filename = utils.path_shorten(filename)
+    local msg
+
+    -- what to show in the last column: filename or symbol information
+    if opts.ignore_filename then
+      msg = vim.api.nvim_buf_get_lines(
+          bufnr,
+          entry.lnum - 1,
+          entry.lnum,
+          false
+        )[1] or ''
+      msg = vim.trim(msg)
```

Comment:
Hmm. I am not sure if i like this. Especially with having a previewer. I think its fine if i can enable and disable this. I do something like this for ctags

Reply:
You can also define the `displayer.items` dynamically based on the opts. I am also doing this with ctags

Reply:
Also for workspace symbols with filename i think something like this would be optimal: `filename:lnum:col ...`

Reply:
I think it's hard to know what the user will want.
I personally avoid previewers where I can, and I'd certainly use it. For document specific finders (lsp_document, treesitter) it makes more sense than having the filename repeated in the column N times.

If it were to be configurable to the point the column would be empty, then I'd do the dynamic sizing and reclaim the space.

Reply:
Agree on the reformatting of the filename:lnum:..

Reply:
I agree that it makes more sense for the `lsp_document` to not have the filename but it would also be awesome to not have the line shown if the user wishes it. I think we can do something like this to realize it
```lua
local displayer_items = {
  { width = 6 }, -- filename and lnum og would be 6 spacing for only lnum:col
  { remaining = true }, -- default for our symbol, using the remaining space
}

if opts.show_filename then
  displayer_items[1].width = 25 -- make it take enough space for filename. Don't know how much.
end

if opts.show_line then
  table.insert(displayer_items, 2, { width = 9 }) -- restrict symbols to 9 width and show_line will now take the remaining space.
end

local displayer = entry_display.create {
  separator = " ",
  items = display_items
}
```

Now if show_line is set then we can construct it and if now we just pass in nil as third element. And it will be ignored. Does this make sense for you. I just like to provide the user with the most amount of freedom to create his picker like he wants to.